core/trino-main/src/main/java/io/trino/connector/informationschema/InformationSchemaPageSource.java

```diff
@@ -285,7 +284,7 @@ private void addTablesRecords(QualifiedTablePrefix prefix)
         }
         // TODO (https://github.com/trinodb/trino/issues/8207) define a type for materialized views
 
-        for (SchemaTableName name : union(tables, views)) {
+        for (SchemaTableName name : tables) {
```

Comment:
what was the reason for union in the first place?

Reply:
I don't know. The code dates back 2014 https://github.com/trinodb/trino/commit/5a727b89373642453e0c39a3301035e1190c28cd. Maybe back then table listing didn't include views, or maybe the code author was simply confused about "list tables" means. I wouldn't be surprised either way, I've seen people being confused and connectors being implemented incorrectly (https://github.com/trinodb/trino/pull/11063). That's why i proposed renaming "list tables" to "list relations". 
See https://github.com/trinodb/trino/pull/19832#discussion_r1399768605